speechbrain/dataio/iterators.py

```diff
@@ -0,0 +1,218 @@
+"""Webdataset compatible iterators
+
+Authors:
+ * Aku Rouhe 2021
+"""
+import bisect
+import random
+from dataclasses import dataclass, field
+from functools import partial
+from typing import Any
+from speechbrain.dataio.batch import PaddedBatch
+
+
+@dataclass(order=True)
+class LengthItem:
+    length: int
+    data: Any = field(compare=False)
+
+
+def total_length_with_padding(lengths):
+    # How long would batch be (with padding)
+    return len(lengths) * max(lengths)
+
+
+def padding_ratio(lengths):
+    # How much of batch is padding:
+    return 1.0 - sum(lengths) / total_length_with_padding(lengths)
+
+
+@dataclass(order=True)
+class RatioIndex:
+    ratio: float
+    index: int
+
```

Comment:
I'm not super familiar with `@dataclass`... what's the rationale for using that here rather than e.g. named tuple or something?

Reply:
Ah, I guess its for the automatic generation of `__lt__` `__gt__` etc.

Reply:
This particular example could have been a namedtuple as well.  The other dataclass, LengthItem, makes it so that only the length value is ever used in comparisons. Since that one was a dataclass, I made this like that too. I think the main difference in usage is that tuples are immutable.